### PR TITLE
Align maven plugin versions to Apache parent pom bumped to version 29

### PR DESCRIPTION
**Pull request by bowenliang123**
### _Why are the changes needed?_

1. Bump Apache parent pom from version `27` to `29`
2. remove duplicate or explicit maven plugin version params, use versions in parent pom instead.

[1] Apache Parent Pom diffs 27 vs 29 https://github.com/apache/maven-apache-parent/compare/apache-27...apache-29
[2] Diff in removed explicit maven plugin version:

maven plugin name | previous explicit version | version in apache pom parent 29 | updated / unchanged
-- | -- | -- | --
antrun | 3.0.0 | 3.1.0 | upgraded
assembly | 3.2.0 | 3.4.2 | upgraded
clean | 3.1.0 | 3.2.0 | upgraded
compiler | 3.8.1 | 3.10.1 | upgraded
dependency | 3.1.1 | 3.4.0 | upgraded
deply | 3.0.0-M1 | 2.8.2 | downgraded
gpg | 3.0.1 | 3.0.1 | upgraded
javadoc | 3.3.0 | 3.4.4 | upgraded
help | 3.2.0 | 3.3.0 | upgraded
jar | 3.2.0 | 3.3.0 | upgraded
shade | 3.4.1 | 3.4.1 | Pinned to 3.4.1 (unchanged)
surefire | 3.0.0-M7 | 2.22.2 | Pinned to 3.0.0-M7 (unchanged)
rat | 0.13 | 0.15 | upgraded
source | 3.2.1 | 3.2.1 | unchanged

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by pan3793** [comment]
It's preferred to manage the `maven-shade-plugin` explicitly since it's bugly, silent version change of this plugin is dangerous. For other plugins, I'm fine to use the version defined in the parent pom.

File: pom.xml [hunk 1]
```diff
@@ -1899,25 +1885,21 @@
...
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-shade-plugin</artifactId>
-                    <version>${maven.plugin.shade.version}</version>
                 </plugin>
 
                 <plugin>
                     <groupId>org.apache.maven.plugins</groupId>
                     <artifactId>maven-jar-plugin</artifactId>
-                    <version>${maven.plugin.jar.version}</version>
                 </plugin>
```
Comment (pan3793) [inline pom.xml:1893]: we can remove the simple definition in the `dependencyManagement` block then.

**Comment by pan3793** [comment]
We can benefit from https://github.com/apache/maven-apache-parent/commit/a7af870e9b3b7a5375f09948ab7fdf9c7709048e to simplify the `pom.xml`

File: pom.xml (hunk 1 above)
Comment (bowenliang123) [reply pom.xml:1893]: Yes, agree.

**Comment by bowenliang123** [comment]
> We can benefit from [apache/maven-apache-parent@a7af870](https://github.com/apache/maven-apache-parent/commit/a7af870e9b3b7a5375f09948ab7fdf9c7709048e) to simplify the `pom.xml`

Sure, let's do the property alignment in the followup PR to make use of them.

**Comment by bowenliang123** [comment]
> It's preferred to manage the `maven-shade-plugin` explicitly since it's bugly, silent version change of this plugin is dangerous. For other plugins, I'm fine to use the version defined in the parent pom.

Pinned `maven-shade-plugin` to 3.4.1.

**Comment by pan3793** [comment]
> > We can benefit from [apache/maven-apache-parent@a7af870](https://github.com/apache/maven-apache-parent/commit/a7af870e9b3b7a5375f09948ab7fdf9c7709048e) to simplify the `pom.xml`
> 
> Sure, let's do the property alignment in the followup PR to make use of them.

After second thought, it's not enough since we can not overwrite `distMgmtReleaseId` and `distMgmtSnapshotsId` after changing, it may cause authentication issues.

**Review by pan3793 — APPROVED** [review]
LGTM, pending CI

**Comment by pan3793** [comment]
Thanks, merged to master

**Pull request closed without merging**